src/check.h.in

```diff
@@ -2261,6 +2261,48 @@ CK_DLL_EXP int CK_EXPORT srunner_has_tap(SRunner * sr);
  */
 CK_DLL_EXP const char *CK_EXPORT srunner_tap_fname(SRunner * sr);
 
+/**
+ * enum describing the specific XML format used for XML logging
+ */
+enum xml_format {
+    CK_XML_FORMAT_UNSPECIFIED,
```

Comment:
`CK_XML_FORMAT_UNSPECIFIED` is the default value when `srunner_create` is called. This is in fact what I was missing and why my tests were failing.

`srunner_xml_format` is used to determine what format to use.
`srunner_xml_format `will never yield `CK_XML_FORMAT_UNSPECIFIED`, it will yield only `CK_XML_FORMAT_DEFAULT` or `CK_XML_FORMAT_JUNIT`.  

The motivation for the unspecified state is that I needed an initial state which was unspecified since the environment variable to set the format only overrides the selected format if it's unspecified.  If it were set to 'default', then I couldn't tell if someone had set it explicitly or not.